### PR TITLE
Created docs for core and rest_api

### PR DESCRIPTION
**Pull request by rds6y9**

### CODE REVIEW

File: ACM_General/rest_api/tests.py [hunk 1]
```diff
@@ -126,7 +150,19 @@ def test_serializer_validation(self):
 
 
 class EventsTestCase(TestCase):
+    """
+    @Desc: Ensures Events behave as expected throughout their lifecycle.
+           This includes all basic CRUD functionality.
```
Comment (kevinschoonover) [inline ACM_General/rest_api/tests.py]: CRUD?

**Review by kevinschoonover — CHANGES_REQUESTED** [review]
Okay, I think the content of the documentation is great, but its is following what I did instead of the restructured text way (my way is bad). I would look at what Zach did in createEvent for better inspiration. The highlights are
1) `@Desc` should just be the top of the docstring i.e. 
    ```
    """"
    Example
    """"
    ```` 
2) All of the parameters should be documented 
3) Only comment about returns if the return exists using the restructured text stuff.

File: ACM_General/rest_api/tests.py (hunk 1 above)
Comment (rds6y9) [reply ACM_General/rest_api/tests.py]: Create, Read, Update, Delete. I was going to reword that lol.

Comment (rds6y9) [reply ACM_General/rest_api/tests.py]: So just to make sure that I understand this correctly, an example should rather be,

"""
Ensures Events behave.....

:param name: what the parameter is
:return: use this only if the return exists
:raise randomError: exception
"""
Correct?

Comment (kevinschoonover) [reply ACM_General/rest_api/tests.py]: """
Ensures Events behave...

:param name: what the paramter is
:type name: str
:rtype: return type (None if none)
:return: I take that back say None here if none, else description.
:raise randomError: exception
"""

**Review by kevinschoonover — APPROVED** [review]

**Pull request merged**